src/git.rs

```diff
@@ -79,40 +115,181 @@ impl Git for LiveGit {
             None => Err(Box::new(git2::Error::from_str("No branch name found")))
         }
     }
-}
 
-#[cfg(test)]
-mod test {
-    use super::*;
-    
-    pub struct MockGit;
-    impl Git for MockGit {
-        fn new() -> Result<Self, Box<dyn Error>> {
-            Ok(MockGit)
+    fn log(&self) -> Result<(), Box<dyn Error>> {
+        let mut revwalk = self.repo.revwalk()?;
+        revwalk.push_head()?;
+        for id in revwalk {
+            let id = id?;
+            let commit = self.repo.find_commit(id)?;
+            Self::debug_commit(&commit);
+        }
+        Ok(())
+    }
+
+    fn find_commit(&self, commit_id: Oid) -> Result<Commit<'_>, Box<dyn Error>> {
```

Comment:
Why not use Option instead of Result here?
Then the code becomes just the RHS of 131.

Reply:
Or rather, if `self.repo.find_commit` produces a Result.  Then why not just call that.  Why the variable?